### PR TITLE
ERC20 as a library.

### PR DESCRIPTION
**Pull request by zie1ony**
This PR provides a new implementation of ERC20 token that is simple and extendable.

See [README.md](https://github.com/casper-ecosystem/erc20/blob/feature/dicts/README.md) for the repository overview.

### CODE REVIEW

**Comment by deuszex** [comment]
Haven't found any technical details that would request a reject of the PR.
Having said that, let's bring up the macro topic on meeting.

**Review by AlexanderLimonov — COMMENTED** [review]
Looks pretty good to me, but do not have time to get in depth into the macro definitions, unfortunately

File: README.md [hunk 1]
```diff
@@ -1,20 +1,290 @@
...
+### ERC20
+
+The `erc20` crate contains the implementation of the ERC20 standard.
+
+#### ERC20 as library
+It can be used as a library to build custom tokens. The code structure allows
+for easy entry points extensions and overrides.
+
+##### Entry Point override example
+Following code shows how to override the `transfer` method to alwasy mint
```
Comment (darthsiroftardis) [inline README.md:35]:
```suggestion
Following code shows how to override the `transfer` method to always mint
```

File: README.md [hunk 2]
```diff
@@ -1,20 +1,290 @@
...
+    * `utils/contract-interface`
+
+The code after the review and addoption should be moved to the separate repo
+and eventualy be added to `casper-contract` and `casper-engine-test-support`. 
+Those functionalities releases a smart contract developer from writing
+boilerplate code and allows for **rapid development**.
+
+#### Test Env Crate
+`utils/test-env` is a small library written on top of 
+`casper-engine-test-support`. It proviedes two structs:
```
Comment (darthsiroftardis) [inline README.md:82]:
```suggestion
`casper-engine-test-support`. It provides two structs:
```

File: README.md [hunk 3]
```diff
@@ -1,20 +1,290 @@
...
+
+#[test]
+fn test_multiple_tokens() {
+    // Prepare the env and users.
+    let env = TestEnv::new();
+    let user1 = env.next_user();
+    let user2 = env.next_user();
+    
+    // Deploy multiple instances of the same contract
+    // agains a single virtual machine.
```
Comment (darthsiroftardis) [inline README.md:127]:
```suggestion
    // against a single virtual machine.
```

**Review by ZeusMonkey — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -237,7 +246,7 @@ fn get_entry_points() -> casper_types::EntryPoints {
     let mut entry_points = casper_types::EntryPoints::new();
     entry_points.add_entry_point(casper_types::EntryPoint::new(
         "constructor",
-        Vec::new(),
+        ::alloc::vec::Vec::new(),
```
Comment (dwerner) [inline README.md:249]: I'd recommend pulling in the `alloc::prelude` as this and several other places follow that idiom as it's used when `std` is enabled. [alloc::prelude](https://doc.rust-lang.org/alloc/prelude/index.html)

Comment (darthsiroftardis) [reply README.md:249]: I think thats coming in due to the macro expansion, and its cause of `cargo expand`

**Comment by zie1ony** [comment]
Closing as macros are not yet ready. Please see #27

File: README.md [hunk 5]
```diff
@@ -1,20 +1,299 @@
...
-$ make test
+```
+make test
+```
+
+## Repository overview
+
+### ERC20
+
+The `erc20` crate contains the implementation of the ERC20 standard.
```
Comment (davidtai) [inline README.md:31]: Should we place holder a cep that this refers to?

**Pull request closed without merging**